common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -481,7 +482,9 @@ void closeAndDeletePartitionsIfNeeded(
     appShuffleInfo.shuffles.forEach((shuffleId, shuffleInfo) -> shuffleInfo.shuffleMergePartitions
       .forEach((shuffleMergeId, partitionInfo) -> {
         synchronized (partitionInfo) {
-          partitionInfo.closeAllFilesAndDeleteIfNeeded(false);
+          AppShufflePartitionInfo.closeAllFilesAndDeleteIfNeeded(false, partitionInfo.dataFile,
+              partitionInfo.getDataChannel(), partitionInfo.getIndexFile(), partitionInfo.getMetaFile(),
+              partitionInfo.getAppAttemptShuffleMergeId(), partitionInfo.getReduceId());
```

Comment:
You can keep `closeAllFilesAndDeleteIfNeeded`, and simply delegate to the new static method from there.
This will also remove the need for `getReduceId` and `getAppAttemptShuffleMergeId`

Reply:
+1